src/reps/rep-utils.js

```diff
@@ -28,6 +28,77 @@ function escapeNewLines(value) {
   return value.replace(/\r/gm, "\\r").replace(/\n/gm, "\\n");
 }
 
+// Map from character code to the corresponding escape sequence.  \0
+// isn't here because it would require special treatment in some
+// situations.  \b, \f, and \v aren't here because they aren't very
+// common.  \' isn't here because there's no need, we only
+// double-quote strings.
+const escapeMap = {
+  // Tab.
+  9: "\\t",
+  // Newline.
+  0xa: "\\n",
+  // Carriage return.
+  0xd: "\\r",
+  // Quote.
+  0x22: "\\\"",
+  // Backslash.
+  0x5c: "\\\\",
+};
+
+/**
+ * Escape a string so that the result is viewable and valid JS.
+ * In particular, control characters, [fixme]
+ */
+function escapeString(str) {
+  let result = "";
```

Comment:
I feel like we should do some performance measurements here to see what kind of impact this method has compared to sanitize string.

Looping over all string characters might be costly.